### PR TITLE
[FE] [MERGE] add rating page & display all comments VisitorWeb

### PR DESCRIPTION
**Pull request by Alban-de-tourtier**
### 🛠 Changes being made

Add the review page and display all the review of the restaurant under the rating form

### 🧪 Test plan

You can test the feature using the "Add a review" button and fill the form and check behind to see it 

### 🏎 PR checklist

1. Read the ticket
- [X] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [X] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [X] is the current version of the destination branch (mostly main) merged into the PR branch?
- [X] does it fulfill the definition of done for the ticket?
- [X] were the tests adjusted or extended for the changed behavior?
- [X] was the documentation adjusted or extended for changed behavior?
- [X] does it align with the team code guidelines?
- [X] check readability of code - is it easy to understand what is happening in the code snippets?
- [X] proper naming
- [X] is it logical?
- [X] are there any performance issues?
- [X] could it be improved / simplified?
4. Execute locally / in testing environment
- [X] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [X] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [X] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [X] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [X] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/visitorWeb/.env [hunk 1]
```diff
@@ -1,2 +1,2 @@
-DB_HOST="http://195.90.210.111:"
+DB_HOST="http://localhost:"
```
Comment (JosefineM2000) [inline packages/visitorWeb/.env:1]: please remove this change, localhost can only be used for local development, not on the server.

File: packages/visitorWeb/src/services/ratingCalls.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import axios from "axios";
+
+const baseUrl = `${process.env.DB_HOST}${process.env.DB_HOST_PORT}/api`;
+
+export const getRatingData = async (name: string) => (
+    axios.get(`${baseUrl}/review/restaurants/${name}`)
+    .then(function (response) {
+        console.log(response.data);
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/services/ratingCalls.ts:8]: please remove log

File: packages/visitorWeb/src/services/ratingCalls.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        return response.data;
+    })
+    .catch(function (error) {
+      console.log(error);
+    }) 
+)
+
+export const postRatingData = async (name: string, comment: string, note: number) => (
+    console.log(comment, note),
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/services/ratingCalls.ts:17]: please remove log

File: packages/visitorWeb/src/services/ratingCalls.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    axios.post(`${baseUrl}/review/restaurants/${name}`, { comment, note })
+    .then(function (response) {
+        console.log(response.data);
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/services/ratingCalls.ts:20]: please remove log

File: packages/visitorWeb/src/components/RestoCard/RestoCard.tsx [hunk 5]
```diff
@@ -117,7 +117,7 @@ const RestoCard = (props: IRestoCardProps) => {
         <Grid item xs={9} className={styles.GridItem}>
           <div className={styles.FlexParent}>
             <h3 className={styles.DishTitle}>{name}</h3>
-            <Rating restoRating={rating} restoRatingsCount={ratingCount} />
+            <Rating restoRating={rating} restoRatingsCount={ratingCount} restoName={name} />
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/RestoCard/RestoCard.tsx:120]: restoRating should display the number and correct average rating in restocard as well, not just on review page

File: packages/visitorWeb/src/pages/RatingPage/RatingPage.tsx [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+  const averageRating = () => {
+    let sum = 0;
+    ratingData.forEach((data) => {
+      sum += data.note;
+    });
+    return parseFloat((sum / ratingData.length).toFixed(1));
+  }
+
+  const addReview = () => {
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/RatingPage/RatingPage.tsx:30]: please add success and failure popups so the user knows if submission was a success

File: packages/visitorWeb/src/pages/RatingPage/RatingPage.tsx
Comment (JosefineM2000) [inline packages/visitorWeb/src/pages/RatingPage/RatingPage.tsx:1]: Please write tests for your new component

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

File: packages/visitorWeb/.env (hunk 1 above)
Comment (Alban-de-tourtier) [reply packages/visitorWeb/.env:1]: yes I will but its for this reason that the .env is always in the .gitignore

File: packages/visitorWeb/src/components/RestoCard/Rating/Rating.tsx [hunk 7]
```diff
@@ -7,59 +7,54 @@ import StarHalfIcon from '@mui/icons-material/StarHalf';
...
-        <ThemeProvider key={index} theme={RatingColor}>
-          <StarOutlineIcon className={styles.StarPosition} color={"primary"} />
-        </ThemeProvider>
-      )}
-      <span className={styles.RatingCount}>
-        {restoRatingsCount}
-      </span>
+    <div className={styles.ReviewContainer}>
+      <Rating name="read-only" value={averageRating()} readOnly />
+      <span className={styles.AverageTxt}>{averageRating()}</span>
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/RestoCard/Rating/Rating.tsx:46]: instead of the average, this should display the number of submitted ratings.

File: packages/restoWeb/src/components/RestoCard/RestoCard.tsx [hunk 8]
```diff
@@ -114,7 +114,7 @@ const RestoCard = (props: IRestoCardProps) => {
         <Grid item xs={9} className={styles.GridItem}>
           <div className={styles.FlexParent}>
             <h3 className={styles.DishTitle}>{resto.name}</h3>
-            <Rating
+            <RatingDisplay
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/RestoCard/RestoCard.tsx:117]: please also update the ratingDisplay in restoWeb, not just visitorWeb to display the correct data (avg rating and number of ratings)

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

File: packages/visitorWeb/cypress/e2e/RatingPageTest/RatingPageTest.cy.ts [hunk 9]
```diff
@@ -0,0 +1,13 @@
+describe('Access to the Rating page', () => {
```
Comment (JosefineM2000) [inline packages/visitorWeb/cypress/e2e/RatingPageTest/RatingPageTest.cy.ts:1]: test is not passing:  CypressError: `cy.click()` can only be called on a single element. Your subject contained 26 elements. Pass `{ multiple: true }` if you want to serially click each element.

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

**Review by JosefineM2000 — APPROVED** [review]
